js/Integer.js

```diff
@@ -0,0 +1,64 @@
+const InvalidInput = require("./Errors")
+
+class Integer {
+    static plus(a, b) {
+        Integer.#validate(a, b)
+
+        let carry = 0
+        while (b !== 0) {
+            carry = a & b
+            a = a ^ b
+            b = carry << 1
+        }
+
+        return a
+    }
+
+    static product(a, b) {
+        Integer.#validate(a, b)
+
+        let signedBit = 1 << 31
+        let isNegativeResult = false
+        if (a & signedBit) {
+            a = Integer.negate(a)
+            isNegativeResult = !isNegativeResult
+        }
+        if (b & signedBit) {
```

Comment:
On 2.: I changed it to signedBitComparator; i think it is misleading to use minInteger because the point of this number is not it's value, but the fact that is has a signed bit and zeroes. But if you prefer minInteger I'll change it.